src/App.tsx

```diff
@@ -1,14 +1,118 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+type TodoWithUser = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user: User,
+};
+
+async function loadTodos() {
+  const todosFromServer = await getTodos();
+
+  return todosFromServer;
+}
+
+async function loadUserForTodo(todo:Todo) {
+  const userFromServer = await getUser(todo.userId);
+
+  return userFromServer;
+}
 
 export const App: React.FC = () => {
+  const [isOpenedModal, setIsOpenedModal] = useState(false);
+  const [openedTodo, setOpenedTodo] = useState<TodoWithUser>(
+    {
+      userId: 1,
+      id: 1,
+      title: 'delectus aut autem',
+      completed: false,
+      user: {
+        id: 1,
+        name: 'Leanne Graham',
+        email: 'Sincere@april.biz',
+        phone: '1-770-736-8031 x56442',
+      },
+    },
+  );
+  const [isDataLoad, setIsDataLoaded] = useState(false);
+  const [isUserLoad, setIsUserLoaded] = useState(false);
+  const [openedTodoId, setOpenedTodoId] = useState(0);
```

Comment:
why do we need `openedTodo` and `openedTodoId `?
can we use only `openedTodo` instead of duplicates?